File: julianoATS/1608_specialArrayWithXElementsGreaterThanOrEqualX.ts
function specialArray(nums: number[]): number {
    const sortedNums: number[] = nums.sort((a,b) => a > b ? a : b);
    const numsLength: number = nums.length;

    for (let index = 1; index < numsLength + 1; index++) {
        let count = 0;
        for (let j = 0; j < numsLength; j++) {
            if (nums[j] >= index) count++;
        }

        if (count === index) return index;
    }

    return -1;
};

File: jehzucco/1608_special_array_with_x_elements_greater_than_or_equal_x.java
class Solution {
    public int specialArray(int[] nums) {

        int x = -1;
        int count = 0;
       
       for(int i = 0; i <= nums.length; i++){
            for(int j = 0; j < nums.length; j++){
                if(nums[j] >= i){
                    count++;
                }
            }
        if(i == count){
                x=i;
        }
        count = 0;
       }
     return x;
    }
}
